### DISCUSSION TITLE
Select input and boolean values

### DISCUSSION THREAD

**Discussion by lucadegasperi** [livewire/livewire#6029] [7. Bugs (archived)]
When combining selects, wire:model and boolean values, the results are not what one would expect.
Here's a demo that reproduces the problem: https://laravelplayground.com/#/snippets/5786417c-fbd0-4499-8bb1-1d5b3c19fb1f
Changing the select won't change the value and the select state is immediately rolled back to the previous one.

I've tried a couple of workarounds but none is in the spirit of simplicity that Laravel and Livewire are known for.

- Using radios is not always a solution as a select can better convey the current state especially when the option value is a long string.

- Using lifecycle hooks to convert the incoming value to a boolean value quickly bloats a component, as more and more fields are added.
```
 public function updatingAttribute(&$value): void
    {
        $value === 'true' ? $value = true : $value = false;
    }
```
- Removing attributes casting from a model and working instead with 0 and 1 values, is not ideal when you need the boolean value elsewhere but I sometimes used this mechanism as a workaround.

The expected behavior is that this should just work, but maybe a modifier like `wire:model.boolean` could help solve this problem?

**Comment by livelogic** [comment]
The modifier has been added as of v3.2.1, although not documented yet.
